### PR TITLE
Split PaletteMap out of Gfx.cpp/Gfx.h

### PR DESCRIPTION
**Pull request by ZehMatt**

### CODE REVIEW

**Comment by LeftofZen** [comment]
Looks like a straight copy-paste of the struct? I'll let the other guys look at it first before I approve it, but it looks good to me.

File: src/OpenLoco/Graphics/PaletteMap.cpp [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    static loco_global<Drawing::PaletteEntry[256], 0x0113ED20> _113ED20;
+
+    const PaletteMap& PaletteMap::getDefault()
+    {
+        static bool initialised = false;
+        static uint8_t data[256];
+        static PaletteMap defaultMap(data);
+        if (!initialised)
+        {
+            std::iota(std::begin(data), std::end(data), 0);
```
Comment (duncanspumpkin) [inline src/OpenLoco/Graphics/PaletteMap.cpp:24]: It was pointed out in openrct2 that we never set the initialised variable to true.

Comment (ZehMatt) [reply src/OpenLoco/Graphics/PaletteMap.cpp:24]: I just moved things around and didn't really look for the details, but I can probably improve that part.

File: src/OpenLoco/Graphics/Gfx.cpp [hunk 2]
```diff
@@ -1924,20 +1803,4 @@ namespace OpenLoco::Gfx
     {
         _currentFontSpriteBase = base;
     }
-
-    static loco_global<Drawing::PaletteEntry[256], 0x0113ED20> _113ED20;
-
-    // 0x004523F4
-    void loadPalette()
```
Comment (duncanspumpkin) [inline src/OpenLoco/Graphics/Gfx.cpp:1931]: I don't think this should be moved it's for the actual rgb palette rather than the u8 palette maps

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**